seed/lib/superperms/orgs/models.py

```diff
@@ -393,21 +393,41 @@ def organization_pre_delete(sender, instance, **kwargs):
 
 @receiver(pre_save, sender=Organization)
 def presave_organization(sender, instance, **kwargs):
-    """For each instance.access_level_names item changed, update the ali.paths
-    """
     if instance.id is None:
         return
 
     previous = Organization.objects.get(pk=instance.id)
     previous_access_level_names = previous.access_level_names
-    alis = AccessLevelInstance.objects.filter(organization=instance)
-    min_len = min(len(previous_access_level_names), len(instance.access_level_names))
+
+    if previous_access_level_names != instance.access_level_names:
+        _assert_alns_are_valid(instance)
+        _update_alis_path_keys(instance, previous_access_level_names)
+
+
+def _assert_alns_are_valid(org):
+    from seed.models import Column
+    alns = org.access_level_names
+
+    if len(set(alns)) != len(alns):  # if not unique
+        raise ValueError("Organiation's access_level_names must be unique.")
+
+    columns_with_same_names = Column.objects.filter(organization=org, display_name__in=alns)
+    if columns_with_same_names.count() > 0:
+        repeated_names = columns_with_same_names.values_list("display_name", flat=True)
```

Comment:
Since Tax Lots and Properties have a few of the same columns, you may want to filter out any duplicates from the ValueError
```
{
    "status": "error",
    "message": "Organiation's access_level_names must not be column names: <QuerySet ['Address Line 1', 'Address Line 1']>"
}
```